src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -33,107 +35,156 @@
         produces = "application/json")
     public ResponseEntity<List<Pet>> importPets(@RequestBody String csv) {
```

Comment:
when you refactor a long method, try to extract (small) methods for each intention right at the beginning of your refactoring session.
you can also have a look here, which is basically a copy of the contents of Martin Fowler's Refactoring book: [https://refactoring.guru/extract-method](https://refactoring.guru/extract-method)

Reply:
Thank you, yes, starting with this makes sense, we apparently kept that for later. Thank you for sharing the link as well.